WordPress/src/main/java/org/wordpress/android/support/SupportWebViewActivity.kt

```diff
@@ -0,0 +1,75 @@
+package org.wordpress.android.support
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.webkit.WebViewAssetLoader
+import androidx.webkit.WebViewAssetLoader.AssetsPathHandler
+import androidx.webkit.WebViewAssetLoader.DEFAULT_DOMAIN
+import androidx.webkit.WebViewAssetLoader.ResourcesPathHandler
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.support.SupportWebViewActivity.OpenChatWidget.ChatDetails
+import org.wordpress.android.ui.WPWebViewActivity
+
+class SupportWebViewActivity : WPWebViewActivity(), SupportWebViewClient.SupportWebViewClientListener {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        toggleNavbarVisibility(false)
+
+        val assetLoader = WebViewAssetLoader.Builder()
+            .addPathHandler("/assets/", AssetsPathHandler(this))
+            .addPathHandler("/res/", ResourcesPathHandler(this))
+            .build()
+
+        configureWebView()
+        mWebView.webViewClient = SupportWebViewClient(this, assetLoader)
+        mWebView.loadUrl("https://$DEFAULT_DOMAIN/assets/support_chat_widget.html")
+    }
+
+    override fun configureWebView() {
+        super.configureWebView()
+        supportActionBar?.hide()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // We don't want any menu items
+        return true
+    }
+
+    override fun onChatSessionClosed() {
+        setResult(RESULT_OK, intent)
+        finish()
+    }
+
+    class OpenChatWidget : ActivityResultContract<ChatDetails, ChatCompletionEvent?>() {
+        override fun createIntent(context: Context, input: ChatDetails) =
+            Intent(context, SupportWebViewActivity::class.java).apply {
+                putExtra(USE_GLOBAL_WPCOM_USER, true)
+                putExtra(AUTHENTICATION_URL, WPCOM_LOGIN_URL)
```

Comment:
Do we need authentication for this widget? Given embeddable chat widgets are meant to be public?

> Bots that are set to private cannot use the embed code currently as you would have to expose your API key to the public

Reply:
Probably not, just using an available method for now, could be changed later if not required.